### PR TITLE
Use onchain VRF to determine minting success

### LINKED ISSUE #4: Use onchain VRF to revert randomly during minting
### Description

Since the focus of this example is conditional execution within the context of batched EVM calls from Cadence, minting successfully will be the mechanism by which the transaction is conditioned. For the sake of demonstration, we'll use onchain VRF via [Cadence Arch precompile](https://developers.flow.com/evm/how-it-works#precompiled-contracts) to fail within some probability.

This probability doesn't have to be overcomplicated, we could simply revert 1/2 times using a modulus, though we'll want to consider using some other mechanism (see the [random example repo](https://github.com/onflow/random-coin-toss)) to get a random in range without bias if we want some other probability.

### PR DESCRIPTION
**Pull request by sisyphusSmiling**
Closes: #4 

## Description

- Adds random failure on minting ERC721 with 50% probability of success
______

For contributor use:

- [x] Targeted PR against `main` branch
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [x] Code follows the [standards mentioned here](https://github.com/onflow/batched-evm-exec-example/blob/master/CONTRIBUTING.md#styleguides).
- [x] Updated relevant documentation 
- [x] Re-reviewed `Files changed` in the Github PR explorer
- [x] Added appropriate labels

### CODE REVIEW

File: solidity/src/MaybeMintERC721.sol [hunk 1]
```diff
@@ -3,6 +3,7 @@ pragma solidity 0.8.24;
 import {ERC721} from "@openzeppelin/contracts/token/ERC721/ERC721.sol";
 import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
 import {Ownable} from "@openzeppelin/contracts/access/Ownable.sol";
+import {CadenceArchUtils} from "onflow/random-coin-toss/solidity/src/CadenceArchUtils.sol";
```
Comment (nvdtf) [inline solidity/src/MaybeMintERC721.sol]: Should we extract `CadenceArchUtils` into a stand-alone repo? It doesn't make sense to add random coin toss as a dependency just to use a utility contract

Comment (sisyphusSmiling) [reply solidity/src/MaybeMintERC721.sol]: Good callout. I'm wondering if we should have a solidity contracts repo similar to [openzeppelin-contracts](https://github.com/OpenZeppelin/openzeppelin-contracts) so consumers can easily install any Flow EVM contracts. Maybe something like `onflow/flow-sol-contracts` What do you think of this approach?

Comment (nvdtf) [reply solidity/src/MaybeMintERC721.sol]: Yeah good idea. Maybe we can call it `flow-sol-utils` vs. `contracts` that sound much more library/dependency/not-for-everyone

Comment (sisyphusSmiling) [reply solidity/src/MaybeMintERC721.sol]: I'll kick that repo off today. The only contracts I can think of adding at the moment are [`CadenceArchUtils.sol`](https://github.com/onflow/random-coin-toss/blob/main/solidity/src/CadenceArchUtils.sol) (library), [`CadenceRancomConsumer.sol`](https://github.com/onflow/random-coin-toss/blob/main/solidity/src/CadenceRandomConsumer.sol) (abstract contract) and [`Xorshift128plus.sol`](https://github.com/onflow/random-coin-toss/blob/main/solidity/src/Xorshift128plus.sol) (library). I'm sure we'll add more over time.

Actually, what do you think about adding the WFLOW contract into that repo?

Comment (nvdtf) [reply solidity/src/MaybeMintERC721.sol]: Yeah we can add WFLOW too

Comment (sisyphusSmiling) [reply solidity/src/MaybeMintERC721.sol]: Created the [onflow/flow-sol-utils](https://github.com/onflow/flow-sol-utils) repo to store these dependencies. Will update this repo to pull deps from there.

Comment (sisyphusSmiling) [reply solidity/src/MaybeMintERC721.sol]: Updated dependencies to pull from the created repo

**Review by nvdtf — APPROVED** [review]

**Pull request merged**